### PR TITLE
[filebeat][threatintel] Ignore bad indicator IPs for MISP fileset

### LINKED ISSUE #29949: When using the threatintel module, and misp data the fields do not accept cidr notation
Operating System: Centos 8.

Version: Filebeat 7.16.3. and Elasticsearch 7.16.3
Module: Threatintel 
Sub Config: Misp.

Depending on the data from the Misp source, some of the ip's could be in cidr notation specifically when using the ip address field, it doesn't accept the cidr notation as a valid ip. 

example field: 

threatintel.indicator.ip

Quick test to check with elasticsearch: 

```

PUT test
{
  "mappings": {
    "properties": {
      "ip": {
        "type": "ip"
      }
    }
  }
}



POST test/_doc
{
  "ip": "192.168.50.0/24"
}


POST test/_doc
{
  "ip": "192.168.50.2"
}


GET test/_search
{
  "query": {
    "term": {
      "ip": {
        "value": "192.168.50.0/24"
      }
    }
  }
}

DELETE test

````

In general the misp data might include cidrs notation and the beat will have to work with that information.

### PR DESCRIPTION
**Pull request by brsolomon-deloitte**
Type of change:

- Bug

## What does this PR do?

Sets `ignore_failure` when attempting to process the MISP `threat.indicator.ip` using `grok`.

## Why is it important?

MISP may send an Event.Attribute.value IP as a CIDR such as 146.88.240.0/24, which is not a valid IP per the Elasticsearch IP data type.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #29949

### CODE REVIEW

**Comment by brsolomon-deloitte** [comment]
@efd6

File: x-pack/filebeat/module/threatintel/misp/ingest/pipeline.yml [hunk 1]
```diff
@@ -249,6 +249,9 @@ processors:
         - "%{IP:threat.indicator.ip}\\|%{NUMBER:threat.indicator.port}"
       ignore_missing: true
       if: "['ip-src|port', 'ip-dst|port'].contains(ctx.misp?.attribute?.type)"
+      # MISP may send a CIDR such as 1.2.3.0/22, which is not a valid Elasticsearch IP data type
+      # https://github.com/elastic/beats/issues/29949
```
Comment (efd6) [inline x-pack/filebeat/module/threatintel/misp/ingest/pipeline.yml:253]: This link is not necessary since it's in the changelog.

**Review by efd6 — COMMENTED** [review]
Are you able to add a test case for this?

**Comment by brsolomon-deloitte** [comment]
> Are you able to add a test case for this?

I am not, due to lack of experience developing against filebeat, but you can use this MISP record as a test input:

```json
{
  "Event": {
    "Attribute": {
      "Galaxy": [],
      "ShadowAttribute": [],
      "category": "Network activity",
      "comment": "",
      "deleted": false,
      "disable_correlation": false,
      "distribution": "5",
      "event_id": "1528",
      "first_seen": null,
      "id": "4081012",
      "last_seen": null,
      "object_id": "0",
      "object_relation": null,
      "sharing_group_id": "0",
      "timestamp": "1670293864",
      "to_ids": true,
      "type": "ip-dst",
      "uuid": "4fa3610f-5412-41dd-9034-c8294fc3c7c2",
      "value": "146.88.240.0/24"
    },
    "CryptographicKey": [],
    "EventReport": [],
    "Galaxy": [],
    "Object": [],
    "Org": {
      "id": "1",
      "local": true,
      "name": "ORGNAME",
      "uuid": "121f5d02-2e2e-4180-9166-fcad7e01a20d"
    },
    "Orgc": {
      "id": "1",
      "local": true,
      "name": "ORGNAME",
      "uuid": "121f5d02-2e2e-4180-9166-fcad7e01a20d"
    },
    "RelatedEvent": [
      {
        "Event": {
          "Org": {
            "id": "1",
            "name": "ORGNAME",
            "uuid": "121f5d02-2e2e-4180-9166-fcad7e01a20d"
          },
          "Orgc": {
            "id": "3",
            "name": "CIRCL",
            "uuid": "55f6ea5e-2c60-40e5-964f-47a8950d210f"
          },
          "analysis": "2",
          "date": "2018-03-26",
          "distribution": "3",
          "id": "896",
          "info": "OSINT - Forgot About Default Accounts? No Worries, GoScanSSH Didnâ€™t",
          "org_id": "1",
          "orgc_id": "3",
          "published": true,
          "threat_level_id": "3",
          "timestamp": "1523865236",
          "uuid": "5acdb4d0-b534-4713-9612-4a1d950d210f"
        }
      }
    ],
    "ShadowAttribute": [],
    "Tag": [
      {
        "colour": "#004577",
        "exportable": true,
        "hide_tag": false,
        "id": "1",
        "is_custom_galaxy": false,
        "is_galaxy": false,
        "local": 0,
        "local_only": false,
        "name": "osint:source-type=\"block-or-filter-list\"",
        "numerical_value": null,
        "user_id": "0"
      }
    ],
    "analysis": "2",
    "attribute_count": "2053",
    "date": "2022-07-26",
    "disable_correlation": false,
    "distribution": "0",
    "event_creator_email": "admin@admin.test",
    "extends_uuid": "",
    "id": "1528",
    "info": "firehol_level1 feed",
    "locked": false,
    "org_id": "1",
    "orgc_id": "1",
    "proposal_email_lock": false,
    "protected": null,
    "publish_timestamp": "0",
    "published": false,
    "sharing_group_id": "0",
    "threat_level_id": "4",
    "timestamp": "1672885819",
    "uuid": "2ed041a1-33ea-4e54-bf26-bf1c7ce191b3"
  }
}
```

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**